src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1321,6 +1321,93 @@ def _validate_trusted_launch(namespace):
+                                           namespace.os_sku, latest_version)
+                vm_image_generation_version = vm_image_info.hyper_v_generation
+            else:
+                vm_image_info = client.get(namespace.location, namespace.os_publisher, namespace.os_offer,
+                                           namespace.os_sku, namespace.os_version)
+                vm_image_generation_version = vm_image_info.hyper_v_generation
+            if vm_image_generation_version == 'V1':
+                logger.warning('Consider upgrading security for your workloads using Azure Trusted Launch VMs.'
+                               ' To know more about Trusted Launch, please visit'
+                               ' https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+            if vm_image_generation_version == 'V2':
+                if vm_image_info.features and _validate_trusted_launch_supported(vm_image_info.features) and\
+                        namespace.security_type is None:
+                    logger.warning('Starting xx/xx/xxxx az vm create command will deploy Trusted Launch VM by default.'
+                                   ' To know more about Trusted Launch, please visit'
+                                   ' https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+
+    if hasattr(namespace, 'attach_os_disk') and namespace.attach_os_disk is not None:
+        from msrestazure.tools import parse_resource_id
+        client = _compute_client_factory(cmd.cli_ctx).disks
+        attach_os_disk_name = parse_resource_id(namespace.attach_os_disk)['name']
+        attach_os_disk_info = client.get(namespace.resource_group_name, attach_os_disk_name)
+        if attach_os_disk_info.hyper_v_generation == 'V1':
+            logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS image'
+                           ' and Trusted Launch security type. To know more about Trusted Launch, please visit'
+                           ' https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+        if attach_os_disk_info.hyper_v_generation == 'V2':
+            if attach_os_disk_info.features and _validate_trusted_launch_supported(attach_os_disk_info.features) and\
+                    namespace.security_type is None:
+                logger.warning('Starting xx/xx/xxxx az vm create command will deploy Trusted Launch VM by default.'
+                               ' To know more about Trusted Launch, please visit'
+                               ' https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+
+
+def _validate_trusted_launch_supported(features):
+    trusted_launch = ['TrustedLaunchSupported', 'TrustedLaunch', 'TrustedLaunchAndConfidentialVmSupported']
+
+    return any(tl_feature in [feature['value'] for feature in features] for tl_feature in trusted_launch)
```

Comment:
```suggestion
def is_trusted_launch_supported(features):
    trusted_launch = ['TrustedLaunchSupported', 'TrustedLaunch', 'TrustedLaunchAndConfidentialVmSupported']

    return any(tl_feature in [feature['value'] for feature in features] for tl_feature in trusted_launch)
```

Reply:
Could we move this method into `_vm_utils.py` so that it can be reused elsewhere?